apps/gitness/src/components-v2/breadcrumbs/project-dropdown.tsx

```diff
@@ -17,7 +19,24 @@ function ProjectDropdown(): JSX.Element {
   const routes = useRoutes()
   const { spaceId } = useParams<PathParams>()
   const navigate = useNavigate()
-  const { spaces } = useAppContext()
+  const { spaces, setSpaces } = useAppContext()
+
+  useEffect(() => {
+    if (!spaces?.length) {
+      membershipSpaces({
+        queryParams: { page: 1, limit: 100, sort: 'identifier', order: 'asc' }
+      })
+        .then(({ body: memberships }) => {
+          const spaceList = memberships.filter(item => item?.space).map(item => item.space as TypesSpace)
+          setSpaces(spaceList)
```

Comment:
doesn't this risk an infinite loop? you're setting state in a useEffect that triggers when the same state changes.
instead, we should move this to a different useEffect, or maybe to the AppContext if possible.

Reply:
Yes definitely, my bad. Moved it to AppContext.